pkg/lib/machine-dialogs.js

```diff
@@ -429,26 +480,35 @@ function AddMachine(dialog) {
         if (!limit || isNaN(limit))
             limit = 20;
 
-        dialog.render({
-            nearlimit : limit * 0.75 <= dialog.machines_ins.list.length,
-            limit : limit,
-            placeholder : _("Enter IP address or host name"),
-            options : invisible,
-        });
+        self.user = { };
 
-        var button = dialog.get_sel(".modal-footer>.pf-m-primary");
-        button.on("click", add_machine);
+        cockpit.user()
+                .done(function (user) {
+                    self.user = user;
+                })
+                .always(function (user) {
+                    dialog.render({
+                        nearlimit : limit * 0.75 <= dialog.machines_ins.list.length,
+                        limit : limit,
+                        placeholder : _("Enter IP address or hostname"),
+                        default_user : self.user.name,
+                        options : invisible,
+                    });
 
-        $("#add-machine-address").on("keyup", function (ev) {
-            if (ev.which === 13) {
-                var disabled = button.prop('disabled');
-                if (!disabled)
-                    add_machine();
-            }
-        });
-        $("#add-machine-address").on("input focus", check_address);
-        color_picker.render("#add-machine-color-picker", dialog.address, color);
-        $("#add-machine-address").focus();
+                    var button = dialog.get_sel(".modal-footer>.pf-m-primary");
+                    button.on("click", add_machine);
+
+                    $("#add-machine-address").on("keyup", function (ev) {
```

Comment:
Let's not do this. Most dialogs don't have auto-action on RET, and doing it for the address will not give people the opportunity to set user and color, which is surprising.

Reply:
Removed.